src/utils/createWeeklyList.ts

```diff
@@ -0,0 +1,162 @@
+import {
+  format,
+  isMonday,
+  previousMonday,
+  isSunday,
+  nextSunday,
+  getUnixTime,
+} from 'date-fns';
+import { MediaInterface, ReportInterface, TotalDataInterface } from 'request';
+
+type CombineDataType = ReportInterface | MediaInterface;
+
+const addTypeAndYearMonthDate = (
+  data: CombineDataType,
+  dataType: 'Report' | 'Media'
```

Comment:
api.d.ts에 `type ApiTypes = 'Report' | 'Media'`요런식으로 선언해놓으면 dataType에서도 사용할 수 있을 것 같습니다
```suggestion
  dataType: ApiTypes
```